### DISCUSSION TITLE
how does kubevela internally store application state?

### DISCUSSION THREAD

**Discussion by nitagarwal** [kubevela/kubevela#6027] [Q&A]
How does kubevela store the application state internally?  How are release artifacts stored?  Does kubevela stores the output of dry run as well as release artifacts?

**Comment by Somefive** [comment]
A bit different from the dry run, KubeVela application records the dispatched manifests in the ResourceTracker object. You can see it through `kubectl get rt`. And the history versions of application are stored in the ApplicationRevision objects.
